python/test/Ice/location/AllTests.py

```diff
@@ -170,7 +177,9 @@ def allTests(helper, communicator):
         test(False)
     except Ice.NotRegisteredException as ex:
         test(ex.kindOfObject == "object")
-        test(ex.id == "unknown/unknown")
+        assert ex.id is not None
+        test(str(ex.id) == "unknown/unknown")
```

Comment:
Why are the `None` check, and the `str` call required here? I think we are missing the type-hint in `NotRegisteredException`

This should be just `test(ex.id == "unknown/unknown")` as before.